pancancer_evaluation/utilities/analysis_utilities.py

```diff
@@ -263,3 +311,58 @@ def get_mad_proportion(coefs, mad_genes):
     except IndexError:
         return 0.0
 
+
+def compute_jaccard(v1, v2):
+    """Compute Jaccard similarity between two lists of terms."""
+    v1, v2 = set(v1), set(v2)
+    intersection = v1.intersection(v2)
+    union = v1.union(v2)
+    return ((len(intersection) / len(union) if len(union) != 0 else 0),
+            len(intersection),
+            len(union))
+
+
+def compare_inter_cancer_coefs(gene_name, per_gene_jaccard, pancancer_comparison_df,
+                               train_set):
+    """Compute average Jaccard similarity between cancer types, for the same gene."""
+    unique_identifiers = list(set(i for i in per_gene_jaccard.keys()))
+    inter_cancer_jaccard = []
+    for id1, id2 in it.combinations(unique_identifiers, 2):
+        coefs_list_1 = per_gene_jaccard[id1]
+        coefs_list_2 = per_gene_jaccard[id2]
+        num_folds = len(coefs_list_1)
+        fold_jaccards = []
+        for f1, f2 in it.product(range(num_folds), repeat=2):
+            try:
+                nz_coefs_1 = list(zip(*coefs_list_1[f1]))[0]
+                nz_coefs_2 = list(zip(*coefs_list_2[f2]))[0]
+                fold_jaccards.append(compute_jaccard(nz_coefs_1, nz_coefs_2)[0])
+            except IndexError:
+                # this can occur if all coefficients for a given fold were zero
+                # (i.e. model predicts the mean/fits only an intercept)
+                # if so, we call it a jaccard index of 0
+                fold_jaccards.append(0.0)
+        try:
+            id1_sig = pancancer_comparison_df[
+                pancancer_comparison_df.identifier == id1
+            ].reject_null.values[0]
+        except IndexError:
+            # if identifier isn't in statistical testing results for some reason,
+            # assume it isn't significant
+            id1_sig = False
```

Comment:
Thanks for pointing this out. After looking into this I do think this is a (somewhat subtle) bug. Basically, the issue is that for genes with very few positive (mutated) samples, sometimes, but not always, the randomly shuffled cross-validation splits will result in a split with samples from only one class.

Right now I'm skipping them in my experiment code, but because the cross-validation splits can vary between seeds, sometimes the model fit works for one seed (present in the analysis) but not another (not present in the pancancer df).

I'm not sure what the best solution to this is. I was aware of this previously and had issue #4 to stratify label proportions in cross-validation, so maybe revisiting that is the best solution. I'll create a future PR if that's what I decide to do, and for now I'll just keep in mind that I could be missing some genes from this analysis.